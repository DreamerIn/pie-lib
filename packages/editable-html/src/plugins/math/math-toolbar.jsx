import { HorizontalKeypad } from '@pie-lib/math-input';

import React from 'react';
import debug from 'debug';
import SlatePropTypes from 'slate-prop-types';
import PropTypes from 'prop-types';

const log = debug('editable-html:plugins:math:math-toolbar');

const toNodeData = data => {
  if (!data) {
    return;
  }

  const { type, value } = data;

  if (type === 'command' || type === 'cursor') {
    return data;
  } else if (value === 'clear') {
    return { type: 'clear' };
  } else {
    return { type: 'write', value };
  }
};

export default class MathToolbar extends React.Component {
  static propTypes = {
    node: SlatePropTypes.node.isRequired,
    value: SlatePropTypes.value.isRequired,
    onChange: PropTypes.func.isRequired
  };

  onClick = data => {
    const { node, value, onChange } = this.props;

    const mathChange = toNodeData(data);

    if (mathChange) {
      const update = { ...node.data.toObject(), change: mathChange };

      log('[send change to node: ', node.key, update);
      const change = value.change().setNodeByKey(node.key, { data: update });
      onChange(change);
    }
  };

  render() {
    return <HorizontalKeypad onClick={this.onClick} />;
  }
}
