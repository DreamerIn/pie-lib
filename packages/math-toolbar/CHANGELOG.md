# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.6.4](https://github.com/pie-framework/pie-lib/compare/@pie-lib/math-toolbar@0.6.2...@pie-lib/math-toolbar@0.6.4) (2019-02-05)


### Bug Fixes

* version bump ([caaf960](https://github.com/pie-framework/pie-lib/commit/caaf960))





## [0.6.2](https://github.com/pie-framework/pie-lib/compare/@pie-lib/math-toolbar@0.6.1...@pie-lib/math-toolbar@0.6.2) (2019-02-05)

**Note:** Version bump only for package @pie-lib/math-toolbar





## [0.6.1](https://github.com/pie-framework/pie-lib/compare/@pie-lib/math-toolbar@0.6.0...@pie-lib/math-toolbar@0.6.1) (2019-01-30)

**Note:** Version bump only for package @pie-lib/math-toolbar





# [0.6.0](https://github.com/pie-framework/pie-lib/compare/@pie-lib/math-toolbar@0.5.1...@pie-lib/math-toolbar@0.6.0) (2019-01-30)


### Features

* libs version bump ([8bedee1](https://github.com/pie-framework/pie-lib/commit/8bedee1))





<a name="0.5.1"></a>
## [0.5.1](https://github.com/pie-framework/pie-lib/compare/@pie-lib/math-toolbar@0.5.0...@pie-lib/math-toolbar@0.5.1) (2019-01-10)


### Bug Fixes

* version bump ([3dcbea9](https://github.com/pie-framework/pie-lib/commit/3dcbea9))





<a name="0.2.1"></a>
## [0.2.1](https://github.com/pie-framework/pie-lib/compare/@pie-lib/math-toolbar@0.2.0...@pie-lib/math-toolbar@0.2.1) (2018-11-27)

**Note:** Version bump only for package @pie-lib/math-toolbar





<a name="0.2.0"></a>
# [0.2.0](https://github.com/pie-framework/pie-lib/compare/@pie-lib/math-toolbar@0.1.1...@pie-lib/math-toolbar@0.2.0) (2018-11-27)


### Features

* **math-toolbar:** changed from ssh to https for the mathquill forked repo ([7364008](https://github.com/pie-framework/pie-lib/commit/7364008))
* **math-toolbar:** updated the mathquill dependency to use the forked repo ([7cc7c06](https://github.com/pie-framework/pie-lib/commit/7cc7c06))





<a name="0.1.1"></a>
## [0.1.1](https://github.com/pie-framework/pie-lib/compare/@pie-lib/math-toolbar@0.1.0...@pie-lib/math-toolbar@0.1.1) (2018-11-27)

**Note:** Version bump only for package @pie-lib/math-toolbar





<a name="0.1.0"></a>
# 0.1.0 (2018-11-23)


### Features

* **math-toolbar:** broke apart editable-html math toolbar into a separate package ([72c5a79](https://github.com/pie-framework/pie-lib/commit/72c5a79))
