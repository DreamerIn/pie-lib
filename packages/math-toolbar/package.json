{
  "name": "@pie-lib/math-toolbar",
  "publishConfig": {
    "access": "public"
  },
  "version": "0.9.3",
  "description": "Math toolbar for editing math equations",
  "keywords": [
    "math",
    "toolbar",
    "equations",
    "expressions"
  ],
  "repository": "pie-framework/pie-lib",
  "main": "lib/index.js",
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@material-ui/core": "^3.8.3",
    "@material-ui/icons": "^3.0.2",
    "@pie-lib/math-input": "^5.1.3",
    "classnames": "^2.2.6",
    "debug": "^4.1.1",
    "keycode": "^2.2.0",
    "mathquill": "git+https://github.com/pie-framework/mathquill-webpack.git",
    "prop-types": "^15.6.2"
  },
  "peerDependencies": {
    "react": "^16.7.0",
    "react-dom": "^16.7.0"
  },
  "devDependencies": {
    "@pie-lib/test-utils": "^0.1.1"
  },
  "gitHead": "392d23fb87a66e7260172db53451a64628b0bf8a"
}
