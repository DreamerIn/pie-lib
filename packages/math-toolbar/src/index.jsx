import React from 'react';
import PropTypes from 'prop-types';
import EditorAndPad from './editor-and-pad';
import { DoneButton } from './done-button';
import { withStyles } from '@material-ui/core/styles';
import MathPreview from './math-preview';

export { MathPreview };

export class MathToolbar extends React.Component {
  static propTypes = {
    latex: PropTypes.string.isRequired,
    onDone: PropTypes.func.isRequired
  };

  constructor(props) {
    super(props);
    this.state = {
      latex: props.latex
    };
  }

  done = () => {
    this.props.onDone(this.state.latex);
  };

  UNSAFE_componentWillReceiveProps(nextProps) {
    this.setState({ latex: nextProps.latex });
  }

  onChange = latex => this.setState({ latex });

  render() {
    const { latex } = this.state;
    return (
      <PureToolbar latex={latex} onChange={this.onChange} onDone={this.done} />
    );
  }
}

export class RawPureToolbar extends React.Component {
  static propTypes = {
    latex: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    onDone: PropTypes.func.isRequired,
    classes: PropTypes.object.isRequired
  };

  render() {
    const { latex, onChange, onDone, classes } = this.props;
    return (
      <div className={classes.pureToolbar}>
        <EditorAndPad latex={latex} onChange={onChange} />
        <DoneButton onClick={onDone} />
      </div>
    );
  }
}
const styles = () => ({
  pureToolbar: {
    display: 'flex',
    width: '100%',
    justifyContent: 'space-between'
  }
});

export const PureToolbar = withStyles(styles)(RawPureToolbar);
