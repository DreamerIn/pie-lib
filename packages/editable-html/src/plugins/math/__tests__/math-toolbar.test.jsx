import { configure, shallow } from 'enzyme';

import { Data, Block, Value } from 'slate';
import { MathToolbar } from '../math-toolbar';
import MockChange from '../../image/__tests__/mock-change';
import React from 'react';
import debug from 'debug';

jest.mock('@pie-lib/math-input', () => {
  return {
    HorizontalKeypad: () => <div>HorizontalKeypad</div>
  };
});

jest.mock('../editor-and-pad', () => () => <div>EditorAndPad</div>);

const log = debug('editable-html:test:editor-and-toolbar');

describe('math-toolbar', () => {
  let onDone;

  beforeEach(() => {
    onDone = jest.fn();
  });

  const mkWrapper = extras => {
    const props = {
      latex: 'foo',
      onDone,
      ...extras
    };

    return shallow(<MathToolbar {...props} />);
  };

  describe('snapshot', () => {
    it('renders', () => {
      const w = mkWrapper();
      expect(w).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    describe('done', () => {
      it('calls onDone', () => {
        const w = mkWrapper();
        w.instance().done();
        expect(onDone).toHaveBeenCalled();
      });
    });
  });
});
