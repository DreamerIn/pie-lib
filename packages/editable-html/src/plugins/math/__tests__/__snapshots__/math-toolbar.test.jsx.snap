// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`math-toolbar snapshot renders 1`] = `
<WithStyles(RawPureToolbar)
  latex="foo"
  onChange={[Function]}
  onDone={[Function]}
/>
`;
